include/p2300/start.hpp

```diff
@@ -0,0 +1,59 @@
+// include/p2300/start.hpp                                            -*-C++-*-
+// ----------------------------------------------------------------------------
+//  Copyright (C) 2022 Dietmar Kuehl http://www.dietmar-kuehl.de
+//
+//  Permission is hereby granted, free of charge, to any person
+//  obtaining a copy of this software and associated documentation
+//  files (the "Software"), to deal in the Software without restriction,
+//  including without limitation the rights to use, copy, modify,
+//  merge, publish, distribute, sublicense, and/or sell copies of
+//  the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be
+//  included in all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+//  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+//  OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+//  NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+//  HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+//  WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+//  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+//  OTHER DEALINGS IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#ifndef INCLUDED_INCLUDE_P2300_START
+#define INCLUDED_INCLUDE_P2300_START
+
+#include <functional.hpp>
+#include <type_traits>
+
+// ----------------------------------------------------------------------------
+// [exec.op_state.start]
+
+namespace std {
+    namespace _Start {
+        class _Cpo {
+        public:
+            template <class _OperationState>
```

Comment:
I don't see a requirement in [the spec](https://brycelelbach.github.io/wg21_p2300_std_execution/std_execution.html#spec-execution.op_state.start) on the `start` CPO that its argument is destructible! The only statically checkable requirements on the argument are that it has a suitable `tag_invoke` overload and that it is an lvalue reference. What is missing, though, are tests that the `start` CPO _can_ be used with non-destructible objects (I will add those). Of course, a corresponding object isn't a viable [`operation_state`](https://brycelelbach.github.io/wg21_p2300_std_execution/std_execution.html#spec-execution.op_state) as this concept requires the entity to be destructible. 

Upon closer reading of [the spec](https://brycelelbach.github.io/wg21_p2300_std_execution/std_execution.html#spec-execution.op_state.start) it is actually not clear what is supposed to happen if the argument is _not_ an lvalue: there is only a statement what should happen if the argument _is_ an lvalue. I also wonder if its use with non-`const` lvalue references is intentional: it could be construed that a `const&` overload for `tag_invoke` is required which would be undesirable. I have raised an [issue](https://github.com/brycelelbach/wg21_p2300_std_execution/issues/440) raising these questions as well as a few others I have about this specific section.

Reply:
That is a remnant from the circular dependency that follows. `operation_state` depends on `execution::start` and execution start requires `operation_state` 

We removed that from the draft, but I am not sure if it is Intentional that `execution::start` could be called with something that is not an operation state